src/sm/launcher/instance.cpp

```diff
@@ -119,18 +119,24 @@ Error Instance::Start()
         return err;
     }
 
+    if (auto err = SetupMonitoring(); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
     auto runStatus = mRunner.StartInstance(mInstanceID, mRuntimeDir, {});
 
+    auto onExit = DeferRelease(&runStatus, [this](runner::RunStatus* runStatus) {
```

Comment:
This is not required because this instance is treated as running and Stop API will be called.